### PR TITLE
Added ability to create and edit contact

### LINKED ISSUE #9: Create  a new contact UI

### PR DESCRIPTION
**Pull request by ubaidchalil**
Fixes #9

### CODE REVIEW

File: app/javascript/src/components/Dashboard/Contacts/Pane/CreateOrEdit.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React from "react";
+
+import { Pane, Typography } from "neetoui";
+
+import Form from "./Form";
+
+const CreateOrEditPane = ({
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/Pane/CreateOrEdit.jsx:7]: Please explain to me your thought process and advantages that you saw while naming the file as `CreateOrEdit.jsx` and naming the component as `CreateOrEditPane`.

File: app/javascript/src/components/Dashboard/Contacts/index.jsx [hunk 2]
```diff
@@ -1,25 +1,81 @@
...
+      contacts[indexOfSelectedContact] = formatContactDataForSave(contact);
+    } else {
+      const sortedContactsById = contacts.sort((a, b) => a.id - b.id);
+
+      const nextId =
+        sortedContactsById.length === 0
+          ? 1
+          : sortedContactsById[sortedContactsById.length - 1].id + 1;
+      contact.id = nextId;
+      contacts.push(formatContactDataForSave(contact));
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/index.jsx:67]: This suggests few things to me:
- You haven't understood how React state hooks work, and the purpose of a state setter
- You haven't understood what mutation is in JavaScript and why it should be avoided
- You are not paying very close attention to the code that you are writing

I want you to carefully go through this line of code and put up a video and point out to me what the issue is over here and also point out what's the right way to have written this code.

You should do this with the highest priority. You can pause all other work that you are doing and put up this video.

File: app/javascript/src/components/Dashboard/Contacts/index.jsx [hunk 3]
```diff
@@ -1,25 +1,81 @@
...
+
+      contacts[indexOfSelectedContact] = formatContactDataForSave(contact);
+    } else {
+      const sortedContactsById = contacts.sort((a, b) => a.id - b.id);
+
+      const nextId =
+        sortedContactsById.length === 0
+          ? 1
+          : sortedContactsById[sortedContactsById.length - 1].id + 1;
+      contact.id = nextId;
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/index.jsx:66]: https://academy.bigbinary.com/learn-intermediate-javascript/mutation/avoid-mutation

**Review by yedhink — COMMENTED** [review]
@ubaidchalil

File: app/javascript/src/components/Dashboard/Contacts/index.jsx (hunk 2 above)
Comment (ubaidchalil) [reply app/javascript/src/components/Dashboard/Contacts/index.jsx:67]: Please check the below video for an explanation of the above issue.

 Let me know your thought on it.

https://www.loom.com/share/d0114498450d471ba87e80e37f09db85

File: app/javascript/src/components/Dashboard/Contacts/Pane/CreateOrEdit.jsx (hunk 1 above)
Comment (ubaidchalil) [reply app/javascript/src/components/Dashboard/Contacts/Pane/CreateOrEdit.jsx:7]: I thought if someone looked into the code other than the file name the purpose would be clear.

**Pull request merged**